### PR TITLE
Fix: CI automations with pre-release versions

### PR DESCRIPTION
**Pull request by aboedo**
Our fastlane automations assumed that version numbers would have the format `x.y.z`. 

If a tag with a different format was made, like a pre-release tag, this would lead to some pretty awkward behavior: 
- the version number was being read from `CFBundleVersion`, which won't include the `-beta.x` suffix
- so when creating a github release, the version for `x.y.z-beta.N` would translate to `x.y.z`
- which would not have a tag, since we actually tagged `x.y.z-beta.N`
- so this would create a new tag from `commitish`... which was set to `master` 🤦. We forgot about that one when we moved to `main`. 
- So that created a tag `3.12.0`, which pointed to the last `master`, `3.4.0`. Sad times. 

Anyway, this fixes that bug by: 

- adding a new `.version` file which will hold the version number, pre-release suffix included if needed. 
- using the version from `.version` for most files, but stripping the pre-release suffix for `.plist` files, since they can't have it. 
- updating `commitish` to the actual git commit. This shouldn't really make a difference, because it should be ignored if a tag already exists. But better safe than sorry.

### CODE REVIEW

File: Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -452,7 +452,7 @@
 			repositoryURL = "https://github.com/RevenueCat/purchases-ios.git";
 			requirement = {
 				kind = upToNextMajorVersion;
-				minimumVersion = 3.9.1;
+				minimumVersion = 3.12.0-SNAPSHOT;
```
Comment (aboedo) [inline Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj:455]: I also updated this and added it to the list of files to automatically update

**Comment by aboedo** [comment]
tests aren't passing because of the stuff I still need to finish in #512, but they're unrelated to this, so it should be safe to merge anyway

File: fastlane/Fastfile [hunk 2]
```diff
@@ -30,13 +30,19 @@ platform :ios do
...
+    
     files_to_update = [
       '../PurchasesCoreSwift.podspec',
       '../Purchases.podspec',
       '../Purchases/Misc/RCSystemInfo.m',
-      '../.jazzy.yaml'
+      '../.jazzy.yaml',
+      '../.version',
+      '../Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj',
+      '../Examples/MagicWeatherSwiftUI/Magic Weather SwiftUI.xcodeproj/project.pbxproj',
```
Comment (vegaro) [inline fastlane/Fastfile:45]: oh nice

File: fastlane/Fastfile [hunk 3]
```diff
@@ -59,14 +65,16 @@ platform :ios do
     rescue
       fail "please add a CHANGELOG.latest.md file before calling this lane"
     end
+    commit_hash = last_git_commit[:commit_hash]
```
Comment (vegaro) [inline fastlane/Fastfile:68]: would this return the last commit in the repository? Or is this equivalent to HEAD?

File: fastlane/Fastfile [hunk 4]
```diff
@@ -240,3 +248,7 @@ def replace_in(previous_text, new_text, path)
   backup_extension = '.bck'
   sh("sed", '-i', backup_extension, sed_regex, path)
 end
+
+def current_version_number
+  File.read("../.version")
```
Comment (vegaro) [inline fastlane/Fastfile:253]: yes, I love this. I was thinking about doing this for the `latest` stuff but ended up not doing it. I like it

File: fastlane/Fastfile (hunk 3 above)
Comment (aboedo) [reply fastlane/Fastfile:68]: from the [source code](https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/fastlane/lib/fastlane/helper/git_helper.rb#L76), this is running `git log -1`, so it should be equivalent to head

File: fastlane/Fastfile (hunk 4 above)
Comment (aboedo) [reply fastlane/Fastfile:253]: definitely makes things easier

**Review by vegaro — APPROVED** [review]

**Pull request merged**